src/test/http_rest_api_handler_test.cpp

```diff
@@ -205,3 +214,252 @@ TEST(ModelControlApi, reloadNotNeededManyThreads) {
+                        {"data_item": "a",
+                        "alias": "new_dummy_output"}
+                    ] 
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                        "data_item": "new_dummy_output"}
+                }
+            ]
+        },
+        {
+            "name": "pipeline2Dummy",
+            "inputs": ["custom_dummy_input"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                            "data_item": "custom_dummy_input"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "a",
+                        "alias": "new_dummy_output"}
+                    ] 
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                        "data_item": "new_dummy_output"}
+                }
+            ]
+        }
+    ]
+})";
+
+TEST(ModelControlApi, configChangeWithPipelines) {
+    std::filesystem::remove("/tmp/ovms_config_file.json");
+    auto configFile = createConfigFileWithContent(config_with_pipeline);
+    char* n_argv[] = {"ovms", "--config_path", "/tmp/ovms_config_file.json", "--file_system_poll_wait_seconds", "0"};
+    int arg_count = 5;
+    ovms::Config::instance().parse(arg_count, n_argv);
+
+    auto handler = ovms::HttpRestApiHandler(10);
+    std::string response;
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    const char* expectedJson_1 = R"("models" : 
+{
+"dummy" : 
+{
+ "model_version_status": [
+  {
+   "version": "1",
+   "state": "AVAILABLE",
+   "status": {
+    "error_code": "OK",
+    "error_message": "OK"
+   }
+  }
+ ]
+}
+},
+"pipelines" : 
+{
+ "pipeline1Dummy" : "AVAILABLE"
+})";
+
+    auto status = handler.processConfigReloadRequest(response);
+    EXPECT_EQ(expectedJson_1, response);
+    EXPECT_EQ(status, ovms::StatusCode::OK_CONFIG_FILE_RELOAD_NEEDED);
+
+    std::filesystem::remove("/tmp/ovms_config_file.json");
+    createConfigFileWithContent(config_with_pipelines);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    const char* expectedJson_2 = R"("models" : 
+{
+"dummy" : 
+{
+ "model_version_status": [
+  {
+   "version": "1",
+   "state": "AVAILABLE",
+   "status": {
+    "error_code": "OK",
+    "error_message": "OK"
+   }
+  }
+ ]
+}
+},
+"pipelines" : 
+{
+ "pipeline1Dummy" : "AVAILABLE",
+ "pipeline2Dummy" : "AVAILABLE"
+})";
+
+    status = handler.processConfigReloadRequest(response);
+    EXPECT_EQ(expectedJson_2, response);
+    EXPECT_EQ(status, ovms::StatusCode::OK_CONFIG_FILE_RELOAD_NEEDED);
+}
+
+TEST(ConfigStatus, configWithPipelines) {
```

Comment:
More explanatory test names -> Test name should briefly tell what the test is doing.